sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -188,45 +174,47 @@ private static void WriteMemory(MemoryStream stream, ReadOnlyMemory<byte> memory
         /// ]]></code>
         /// </example>
         /// <remarks> See error response code and error response message for more detail. </remarks>
-        public virtual Response<UploadLogsResult> Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(ruleId, nameof(ruleId));
             Argument.AssertNotNullOrEmpty(streamName, nameof(streamName));
             Argument.AssertNotNullOrEmpty(logs, nameof(logs));
 
             using var scope = ClientDiagnostics.CreateScope("LogsIngestionClient.Upload");
 
-            RequestContext requestContext = GenerateRequestContext(cancellationToken);
             Response response = null;
-            List<UploadLogsError> errors = new List<UploadLogsError>();
+            var exceptions = new List<Exception>();
+
             scope.Start();
-            try
+            int batchCount = 0; // Keep track of the number of batches as they are uploaded
+
+            // Partition the stream into individual blocks
+            foreach (BatchedLogs<T> batch in Batch(logs, options))
             {
-                // Partition the stream into individual blocks
-                foreach (BatchedLogs<T> batch in Batch(logs, options))
+                batchCount++;
+                try
                 {
                     // Because we are uploading in sequence, wait for each batch to upload before starting the next batch
-                    BatchUpload task = CommitBatchListSyncOrAsync(
+                    response = CommitBatchListSyncOrAsync(
```

Comment:
I've been curious about the naming of the method. Would it make more sense to rename it as UploadBatchSyncOrAsync, or just UploadBatchAsync (the async parameter should convey that this can actually be run synchronously)?